lib/maxitest/verbose_backtrace.rb

```diff
@@ -6,30 +6,31 @@ def self.filter(backtrace)
       end
     end
 
-    class << self
-      attr_accessor :verbose
+    # when printing messages print more if verbose was enabled
+    module Assertion
+      def message
+        if Maxitest::VerboseBacktrace.enabled
+          "#{self.class}: #{super}\n    #{backtrace.join "\n    "}"
+        else
+          super
+        end
+      end
     end
 
-    def plugin_maxitest_verbose_backtrace_init(options)
-      return unless options[:verbose]
-      Maxitest::VerboseBacktrace.verbose = true
-      Minitest.backtrace_filter = Maxitest::VerboseBacktrace::NullFilter
-      Rails.backtrace_cleaner.remove_silencers! if defined?(Rails) && Rails.respond_to?(:backtrace_cleaner)
+    module MinitestPlugin
+      def self.minitest_plugin_init(options)
+        return unless options[:verbose]
+        Maxitest::VerboseBacktrace.enabled = true
+        Minitest.backtrace_filter = Maxitest::VerboseBacktrace::NullFilter
+        # rails has --backtrace which disables rails own backtrace cleaner, but not minitests
+        Rails.backtrace_cleaner.remove_silencers! if defined?(Rails) && Rails.respond_to?(:backtrace_cleaner)
```

Comment:
``
return unless defined?(Rails)
Rails.backtrace_cleaner.remove_silencers!

```

All rails version support it.

Reply:
if rails is just used as gem it might backfire 🤞